src/components/Experience.ts

```diff
@@ -25,15 +40,103 @@ export class Experience<TGameStartr extends FullScreenPokemon> extends Component
     }
 
     /**
-     * Gives experience to a specified pokemon
+     * Gives experience to a specified pokemon.
      *
      * @param pokemon   The pokemon to give experience to.
      * @param experience   The amount of experience to give.
      */
     public gainExperience(pokemon: IPokemon, experience: number): void {
         pokemon.experience += experience;
-        if (pokemon.experience >= this.gameStarter.equations.experienceStarting(pokemon.title, pokemon.level + 1)) {
+        if (pokemon.experience >= this.gameStarter.equations.experienceStarting(pokemon.title, pokemon.level + 1)
+            && pokemon.statistics.health.current !== 0) {
```

Comment:
Why add this check here? If a Pokemon won't level up if while fainted, that means there shouldn't be any ways to give a Pokemon experience when it's fainted. IMO whoever calls `gainExperience` should only do so if the Pokemon isn't fainted.